src/clj/fluree/db/api/query.cljc

```diff
@@ -43,13 +47,20 @@
 (defn restrict-db
   [db t context opts]
   (go-try
-    (let [db*  (if-let [policy-identity (perm/parse-policy-identity opts context)]
-                 (<? (perm/wrap-policy db policy-identity))
-                 db)
-          db** (-> (if t
-                     (<? (time-travel/as-of db* t))
-                     db*))]
-      (assoc-in db** [:policy :cache] (atom {})))))
+    (let [policy-db      (if-let [policy-identity (perm/parse-policy-identity opts context)]
+                           (<? (perm/wrap-policy db policy-identity))
+                           db)
+          time-travel-db (-> (if t
+                               (<? (time-travel/as-of policy-db t))
+                               policy-db))
```

Comment:
Love these more descriptive var names!

Reply:
With the new policy PR, this has changed a little bit so the above wouldn't be compatible. Now you compose dbs, instead of a bunch of if conditions in the API. In this case if you wanted a db to be policy-wrapped, you'd call that API first explicitly. This no longer looks for this option in a regular query.

Also if a query is wrapped in a verifiable credential, there is a new explity query API for that... so this check won't be done always.

Reply:
It looks like that PR has failing checks, should I wait for it to be merged and then adjust this PR to work with it?